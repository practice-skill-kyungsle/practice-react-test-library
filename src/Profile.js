import React from "react";

const Profile = ({ username, name }) => {
    return (
        <div>
            <p>{username}</p>
            <span>({name})</span>
        </div>
    );
};

export default Profile;
