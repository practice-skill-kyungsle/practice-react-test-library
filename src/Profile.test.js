import React from "react";
import { render, screen } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile />", () => {
    it("matches snapshot", () => {
        const utils = render(<Profile username="kyungsle" name="이경수" />);
        expect(utils.container).toMatchSnapshot();
    });
    it("shows the props correctly", () => {
        render(<Profile username="kyungsle" name="이경수" />);
        const linkElement = screen.getByText(/kyungsle/i);
        expect(linkElement).toBeInTheDocument();
    });
});
