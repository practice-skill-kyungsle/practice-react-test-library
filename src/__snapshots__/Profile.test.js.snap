// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <p>
      kyungsle
    </p>
    <span>
      (
      이경수
      )
    </span>
  </div>
</div>
`;
